src/renderer/components/opds/BrowserResult.tsx

```diff
@@ -16,82 +16,98 @@ import {
+
     public componentDidUpdate(prevProps: BrowserResultProps) {
-        if (prevProps.url !== this.props.url || prevProps.location.search !== this.props.location.search) {
+        if (prevProps.url !== this.props.url ||
+            prevProps.location.search !== this.props.location.search) {
             // New url to browse
             this.browseOpds();
         }
     }
 
-    public render(): React.ReactElement<{}>  {
-        const { result } = this.props;
-        let content = (<Loader/>);
-        if (result && result.isFailure) {
-            // browse error
-        } else if (result) {
-            switch (result.data.type) {
-                case OpdsResultType.NavigationFeed:
-                    content = (
-                        <EntryList entries={ result.data.navigation } />
-                    );
-                    break;
-                case OpdsResultType.PublicationFeed:
-                    content = (
-                        <EntryPublicationList publications={ result.data.publications } />
-                    );
-                    break;
-                case OpdsResultType.Empty:
-                    // TRANSLATE
-                    content = (
-                        <Empty />
-                    );
-                    break;
-                default:
-                    break;
+    public render(): React.ReactElement<{}> {
+        const { result, resultIsReject, __ } = this.props;
+        let content = (<Loader />);
+        if (typeof result === "string" && resultIsReject) {
+            content = (
+                <MessageOpdBrowserResult title={__("opds.network.reject")} message={result} />
+            );
+        } else if (typeof result === "object" && result) {
+            if (result.isSuccess) {
+                switch (result.data.type) {
+                    case OpdsResultType.NavigationFeed:
+                        content = (
+                            <EntryList entries={result.data.navigation} />
+                        );
+                        break;
+                    case OpdsResultType.PublicationFeed:
+                        content = (
+                            <EntryPublicationList publications={result.data.publications} />
+                        );
+                        break;
+                    case OpdsResultType.Empty:
+                        content = (
+                            <MessageOpdBrowserResult title={__("opds.empty")} />
+                        );
+                        break;
+                    default:
+                        break;
+                }
+            } else if (result.isTimeout) {
+                content = (
+                    <MessageOpdBrowserResult title={__("opds.network.timeout")} />
+                );
+            } else {
+                content = (
+                    <MessageOpdBrowserResult
+                        title={`${__("opds.network.error")}${result.statusCode} ${result.statusMessage}`}
```

Comment:
There is no space character after the translated `opds.network.error` message (before the HTTP status code).
That being said, I think this is a prime candidate for `i18next` interpolation, such as:

https://github.com/readium/readium-desktop/blob/5a235e99460f162c887c90840473dbc8558f1272/src/resources/locales/en.json#L93-L100

https://www.i18next.com/translation-function/interpolation

Reply:
i changed the text print
I added a message field